support/hololens/ServoApp/BrowserPage.cpp

```diff
@@ -252,15 +217,20 @@ void BrowserPage::Flush() {
 void BrowserPage::MakeCurrent() { mOpenGLES.MakeCurrent(mRenderSurface); }
 
 void BrowserPage::WakeUp() {
-  // FIXME: this won't work if it's triggered while the thread is not
-  // waiting. We need a better looping logic.
-  HANDLE hEvent = ::OpenEventA(EVENT_ALL_ACCESS, FALSE, sWakeupEvent);
-  ::SetEvent(hEvent);
+  EnterCriticalSection(&mGLLock);
+  mPendingWakeup = true;
```

Comment:
How about just doing `RunOnGLThread([=] { });`, so that the flag can be removed? 

And it could also be a hook to provide a specific event to `WakeUp()`, where the match would happen here and the "right" task would be added to the queue?

Reply:
If you use `RunOnGLThread([=] { })`, you also don't need to use the critical section and wake stuff here...

Reply:
That what I was initially doing. Just felt like capturing the closure context on WakeUp was an unnecessary overhead. But it's probably minimal. For the sake of simplicity I might get rid of mPendingWakeup.

Reply:
ok